### PR TITLE
Delay announcement of posts within a given window (closes #47)

### LINKED ISSUE #47: Only announce posts that are over N minutes old

### PR DESCRIPTION
**Pull request by n7st**
# Changelog

## Added

* `reddit.minimum_post_age` config item. This is a duration (10s, 1m, 1h etc) which controls ignoring posts under a certain age. This gives Automod time to clean up spam posts.

## Changed

* The `reddit.poll_wait_time`  config item is now called `reddit.poll_wait_duration`, and accepts a duration (10s, 1m, 1h etc).
* The reddit API client now requires a real, registered application and its configuration. Instructions can be found in the README. This is to avoid the rate-limiting problems which come with using a generic client with a generic user-agent.

### CODE REVIEW

File: internal/app/subpoll/poller.go [hunk 1]
```diff
@@ -127,3 +151,20 @@ func (p *Poller) setLastPollTime() {
...
 }
+
+func (p *Poller) postIsOldEnough(post *reddit.Post) bool {
+	var minimumPostAgeSeconds = p.Config.Reddit.MinimumPostAge
+
+	if minimumPostAgeSeconds == 0 {
+		return true
+	}
+
+	oldEnough := time.Since(post.Created.Time) > p.Config.Reddit.MinimumPostAge
```
Comment (cj123) [inline internal/app/subpoll/poller.go:162]: can't you use the `minimumPostAgeSeconds` variable here? also minimumPostAgeSeconds isn't really a valid variable name now given that it's a time.Duration?

File: internal/app/subpoll/poller.go [hunk 2]
```diff
@@ -41,6 +41,8 @@ func Load(config *config.Config, bot *subgrok.Bot) *Poller {
 		Subscriptions: &subscription.Subscriptions{},
 	}
 
+	poller.TooRecent = make(map[string]bool)
```
Comment (cj123) [inline internal/app/subpoll/poller.go:44]: is there a reason this isn't initialised in the poller struct initialisation above?

File: internal/app/subpoll/poller.go (hunk 1 above)
Comment (n7st) [reply internal/app/subpoll/poller.go:159]: Resolved in 30c17e5.

File: internal/app/subpoll/poller.go (hunk 2 above)
Comment (n7st) [reply internal/app/subpoll/poller.go]: Resolved in 30c17e5.

**Pull request merged**